src/PhpSpreadsheet/Reader/Xlsx.php

```diff
@@ -368,6 +368,9 @@ private function getFromZipArchive(ZipArchive $archive, $fileName = '')
         if (strpos($fileName, '//') !== false) {
             $fileName = substr($fileName, strpos($fileName, '//') + 1);
         }
+        // Relative paths generated by dirname($filename) when $filename
+        // has no path (i.e.files in root of the zip archive)
+        $fileName = (string) preg_replace('/^\.\//', '$1', $fileName);
```

Comment:
Sorry for not noticing this earlier. Your regular expression works, but only accidentally. The value in `$1` should never be set given the regular expression you're using. I think setting the replacement value to the null string is what you want to do.

Reply:
Absolutely. I simplified the former regex (which grouped apart the path), and left the replacement that happens to be always null. It's already fixed.